webdriver/webdriver.go

```diff
@@ -13,28 +15,71 @@ var (
 	startFrameBuffer = flag.Bool("frame_buffer", frameBufferDefault(), "Whether to use a frame buffer")
 	seleniumPath     = flag.String("selenium_path", "", "Path to the selenium standalone binary.")
 	seleniumHost     = flag.String("selenium_host", "localhost", "Host to run selenium on")
-	seleniumPort     = flag.Int("selenium_port", 8888, "Port to run selenium on")
+	seleniumPort     = flag.Int("selenium_port", 0, "Port to run selenium on")
 )
 
 func frameBufferDefault() bool {
 	return runtime.GOOS != "darwin"
 }
 
+func pickUnusedPort() int {
```

Comment:
Do I understand correctly that this is kind of "best effort"? There's nothing that prevents a race condition where another unrelated process acquires the port after `l.Close()` and before the intended subprocess has a chance to acquire it.

Reply:
Yes, it's best-effort. The details are a bit nuanced but it generally works:

Closing a TCP socket puts it into TIME_WAIT, which has a short timeout (~2min) during which the kernel won't reuse the port. However, if a program specifically asks to bind that port, the kernel will allow that to happen as there is no active listening socket on the port. In addition, kernel assigns ephemeral ports pseudo-randomly to mitigate some types of attacks. However, a user/program could accidentally/maliciously race to bind to the same port explicitly.

In our case, the likelihood is low and acceptable. Besides, I try to use this trick as little as possible (only use it when we need to know the port number or the subprocess doesn't support binding to 0).

Reply:
Yeah, I found a good article that can serve as explanation and will link to it.